fe/fe-core/src/main/cup/sql_parser.cup

```diff
@@ -1419,6 +1419,10 @@ alter_stmt ::=
     {:
         RESULT = new AlterResourceStmt(resourceName, properties);
     :}
+    | KW_ALTER KW_STORAGE KW_VAULT ident_or_text:vaultName opt_properties:properties
+    {:
+        RESULT = new AlterStorageVaultStmt(vaultName, properties);
+    :}
```

Comment:
since we will remove legacy planner soon, please impl it in nereids directly, and do not create a new class in doris/analysis